src/edu/csus/ecs/pc2/clics/API202306/CLICSLanguage.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 1989-2024 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.clics.API202306;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.csus.ecs.pc2.core.LanguageUtilities;
+import edu.csus.ecs.pc2.core.model.Language;
+import edu.csus.ecs.pc2.core.util.IJSONTool;
+import edu.csus.ecs.pc2.services.core.JSONUtilities;
+
+/**
+ * CLICS Language
+ * Contains information about a language
+ *
+ * @author John Buck
+ *
+ */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CLICSLanguage {
+
+    @JsonProperty
+    private String id;
+
+    @JsonProperty
+    private String name;
+
+    @JsonProperty
+    private boolean entry_point_required;
+
+    @JsonProperty
+    private String entry_point_name;
+
+    @JsonProperty
+    private String [] extensions;
+
+    @JsonProperty
+    private CLICSCommand compiler;
+
+    @JsonProperty
+    private CLICSCommand runner;
+
+    /**
+     * Fills in the test case properties
+     *
+     * @param language The language being serialized
+     */
+    public CLICSLanguage(Language language) {
+        id = IJSONTool.getLanguageId(language);
+        name = language.getDisplayName();
```

Comment:
I suspect that `getDisplayName()` is not the best first choice for assigning to the CLICS language `name` field.  I say this because we frequently (by default, in fact) define (for example) the DisplayName for Java as `Java` -- but the CLICS spec indicates (recommends) that the `name` for the Java language be `java`  (all lower-case).

I think it might make sense to first call `language.getId()` (which I'm pretty sure is intended to return a CLICS ID) and use that if it's non-null && non-empty, _otherwise_ use `getDisplayName()`.

Reply:
Then the '`id`' would be the same as the '`name`'? I think the CLICS spec says the name for java should "Java".  Maybe you were looking at the wrong column?

Reply:
Yep, my incorrect reading of the spec.